server/komodo-rest/src/main/java/org/komodo/rest/service/KomodoMetadataService.java

```diff
@@ -3579,13 +3579,28 @@ public Response publishVirtualization(final @Context HttpHeaders headers, final
             Repository repo = this.kengine.getDefaultRepository();
             uow = createTransaction(principal, "publish", true); //$NON-NLS-1$
             Vdb vdb = findVdb(uow, attributes.getName());
+            if (vdb == null) {
+                //
+                // We don't have a vdb so maybe we have a dataservice instead
+                // Find the dataservice's vdb to publish.
+                //
+                Dataservice dataservice = findDataservice(uow, attributes.getName());
```

Comment:
In the case of user having a same name vdb and dataservice in their repo, maybe the user intended to publish the dataservice.  There is no way to do that with this method, since it looks for a vdb with that name first.  Maybe also need to provide the type in the request to alleviate that?

Reply:
There should be no DataService in Beetle IMO, we need weed it out as that is what we did in DSB and I do not think that will fit here. It causes more confusion than anything else.

Reply:
The reason we created dataservice object initially was to encapsulate everything needed to make the service portable to another server.  That was well before our openshift service strategy.  So yeah, we can probably move away from using it at this time.  This will require some redesign and is a topic for future sprint(s).

Reply:
@mdrillin Add a JIRA, then we can identify first what we can put on the chopping block

Reply:
TEIIDTOOLS-399